File: individual/client/src/main/java/fields/ErrorCode.java
package fields;

public class ErrorCode {
    /** Not defined, see error message (if any). **/
    static final short NOT_DEFINED = 0;

    /** File not found. **/
    static final short FILE_NOT_FOUND = 1;

    /** Access violation. **/
    static final short ACCESS_VIOLATION = 2;

    /** Disk full or allocation exceeded. **/
    static final short MEMORY_ALLOCATION = 3;

    /** Illegal TFTP operation. **/
    static final short ILLEGAL_OPERATION = 4;

    /** Unknown transfer ID. **/
    static final short UNKNOWN_TID = 5;

    /** File already exists. **/
    static final short FILE_EXISTS = 6;

    /** No such user. **/
    static final short NO_SUCH_USER = 7;

    static final String[] definition = {
            "Not defined, see error message (if any)",
            "File not found",
            "Access violation",
            "Disk full or allocation exceeded",
            "Illegal TFTP operation",
            "Unknown transfer ID",
            "File already exists",
            "No such user"
    };
}

File: udp/client/tftpclient/src/main/kotlin/ru/spbau/smirnov/tftp/client/ErrorCode.kt
package ru.spbau.smirnov.tftp.client

enum class ErrorCode(val code: Int) {
    NOT_DEFINED(0),
    FILE_NOT_FOUND(1),
    ACCESS_VIOLATION(2),
    DISK_FULL(3),
    ILLEGAL_OPERATION(4),
    UNKNOWN_TRANSFER_ID(5),
    FILE_EXISTS(6),
    NO_SUCH_USER(7);

    companion object {
        fun byCode(code: Int): ErrorCode {
            return when (code) {
                0 -> NOT_DEFINED
                1 -> FILE_NOT_FOUND
                2 -> ACCESS_VIOLATION
                3 -> DISK_FULL
                4 -> ILLEGAL_OPERATION
                5 -> UNKNOWN_TRANSFER_ID
                6 -> FILE_EXISTS
                7 -> NO_SUCH_USER
                else -> throw IllegalArgumentException("No such code")
            }
        }
    }
}